### PR TITLE
add private subnets

### PR DESCRIPTION
**Pull request by melema123**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -66,3 +66,28 @@ variable "cidr_new_bits" {
...
+  common_tags = {
+    Name            = "${var.vpc_name}-${var.env}"
+    RawName         = var.vpc_name
+    OwnerList       = var.owner
+    EnvironmentList = var.env
+    EndDate         = var.end_date
+    ProjectList     = var.project
+    DeploymentType  = var.deployment_type
+  }
+}
```
Comment (morrismukiri) [inline variables.tf:93]:
```suggestion
}

```

File: nat-gatway.tf [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  vpc   = true
+  tags  = local.common_tags
+}
+
+resource "aws_nat_gateway" "nat_gateway" {
+  count         = var.allow_private_subnets_access_to_internet ? 1 : 0
+  allocation_id = aws_eip.ip[count.index].id
+  subnet_id     = aws_subnet.onadata-api-subnets[count.index].id
+  tags          = local.common_tags
+}
```
Comment (morrismukiri) [inline nat-gatway.tf:12]:
```suggestion
}

```

File: nat-gatway.tf [hunk 3]
```diff
@@ -0,0 +1,12 @@
+resource "aws_eip" "ip" {
+  count = var.allow_private_subnets_access_to_internet ? 1 : 0
+  vpc   = true
+  tags  = local.common_tags
+}
+
+resource "aws_nat_gateway" "nat_gateway" {
+  count         = var.allow_private_subnets_access_to_internet ? 1 : 0
```
Comment (morrismukiri) [inline nat-gatway.tf:8]: Can we add name tag?

File: nat-gatway.tf [hunk 4]
```diff
@@ -0,0 +1,12 @@
+resource "aws_eip" "ip" {
```
Comment (morrismukiri) [inline nat-gatway.tf:1]: Can we name it something that identifies this as specific to the private subnet? 

e.g 
```
resource "aws_eip" "privates_subnet_nat_gateway_public_ip" 
```

File: network.tf [hunk 5]
```diff
@@ -85,18 +49,21 @@ resource "aws_route_table" "without-peer" {
...
 
-  tags = {
-    Name            = "${var.vpc_name}-${var.env}"
-    RawName         = var.vpc_name
-    OwnerList       = var.owner
-    EnvironmentList = var.env
-    EndDate         = var.end_date
-    ProjectList     = var.project
-    DeploymentType  = var.deployment_type
+resource "aws_route_table" "connection_to_nat_gatway" {
```
Comment (morrismukiri) [inline network.tf:55]: Can we name it something that identifies this as specific to the private subnet? 

e.g 

```suggestion
resource "aws_route_table" "private_subnet_connection_to_nat_gatway" {
```

Comment (melema123) [reply network.tf:55]: @morrismukiri updated as requested. Please review again.

**Pull request closed without merging**